server/raft_test.go

```diff
@@ -1303,49 +1303,13 @@ func TestNRGCatchupDoesNotTruncateCommittedEntriesDuringRedelivery(t *testing.T)
 
 	// Simulate receiving an old entry as a redelivery. We should not truncate as that lowers our commit.
 	n.processAppendEntry(aeMsg1, n.aesub)
-	require_Equal(t, n.commit, 1)
+	require_Equal(t, n.commit, 2)
 
 	// Heartbeat, makes sure we commit.
 	n.processAppendEntry(aeHeartbeat2, n.aesub)
 	require_Equal(t, n.commit, 3)
 }
 
-func TestNRGNewLeaderWithIncorrectPtermDoesNotTruncateIfCommitted(t *testing.T) {
```

Comment:
These two tests that have been removed asserted that `n.pterm = n.term` could still be executed, resulting in a situation where we should recover from the possibility of having an incorrect `pterm`.

Since its removal in https://github.com/nats-io/nats-server/pull/6060, these tests should be removed. We should never have this situation occur anymore.